emily/handler/src/api/handlers/chainstate.rs

```diff
@@ -171,22 +180,28 @@ pub async fn update_chainstate(
 /// TODO(TBD): Consider moving this logic into database accessor structures.
 pub async fn add_chainstate_entry_or_reorg(
     context: &EmilyContext,
+    can_reorg: bool,
     chainstate: &Chainstate,
 ) -> Result<(), Error> {
     // Get chainstate as entry.
     let entry: ChainstateEntry = chainstate.clone().into();
     debug!("Attempting to add chainstate: {entry:?}");
     match accessors::add_chainstate_entry_with_retry(context, &entry, 15).await {
         Err(Error::InconsistentState(Inconsistency::Chainstates(conflicting_chainstates))) => {
-            info!("Inconsistent chainstate found; attempting reorg for {entry:?}");
-            let execute_reorg_request = ExecuteReorgRequest {
-                canonical_tip: chainstate.clone(),
-                conflicting_chainstates,
-            };
-            // Execute the reorg.
-            execute_reorg_handler(context, execute_reorg_request)
-                .await
-                .inspect_err(|e| warn!("Failed executing reorg with error {}", e))?;
+            if can_reorg {
+                info!("Inconsistent chainstate found; attempting reorg for {entry:?}");
+                let execute_reorg_request = ExecuteReorgRequest {
+                    canonical_tip: chainstate.clone(),
+                    conflicting_chainstates,
+                };
+                // Execute the reorg.
+                execute_reorg_handler(context, execute_reorg_request)
+                    .await
+                    .inspect_err(|e| warn!("Failed executing reorg with error {}", e))?;
+            // Log error.
+            } else {
+                info!("Inconsistent chainstate found for {entry:?} but we pretend it's okay.");
```

Comment:
I'd probably change this to like trace or debug since it'll probably happen a lot?

Reply:
Okay

Reply:
Done.